### PR TITLE
Refactoring of VS projects, start using nuget dependencies for OpenSSL and zlib on Windows.

### PR DESCRIPTION
**Pull request by jtattermusch**
-- started using NuGet packages for zlib and openssl dependencies
-- we have only one set of solution and project files that supports VS2013,VS2013 and VS2010.
-- stripped lot of useless junk
-- added a README.md for vsprojects

### CODE REVIEW

**Comment by ctiller** [comment]
I raced with this - we might need to chat to figure out how to merge.

**Comment by nicolasnoble** [comment]
Did you two guys talk ? Or is everything clear now ?

**Comment by jtattermusch** [comment]
Seems like I was able to merge Grpc.mak.template without trouble and it contains Craig's changes.

**Comment by nicolasnoble** [comment]
Alright, cool.

**Comment by ctiller** [comment]
Please test run_tests.py at least starts a build before merging this.

**Comment by jtattermusch** [comment]
Fixed run_tests.py. It doesn't build all the tests, but it builds quite a few.

File: third_party/openssl [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 3df69d3aefde7671053d4e3c242b228e5d79c83f
+Subproject commit 4ac0329582829f5378d8078c8d314ad37db87736
```
Comment (nicolasnoble) [inline third_party/openssl]: Please don't revert OpenSSL :-)

File: vsprojects/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+This directory contains MS Visual Studio project & solution files.
+
+#Supported Visual Studio versions
+
+Currently supported versions are Visual Studio 2013 (our primary focus) and 2010.
+
+#Building
+We are using [NuGet](http://www.nuget.org) to pull zlib and openssl dependencies.
+If you don't have Visual Studio NuGet plugin installed, you'll need to
+download nuget.exe from web and manually restore the NuGet packages.
```
Comment (nicolasnoble) [inline vsprojects/README.md]: from *the* web, I guess ?

**Comment by jtattermusch** [comment]
Reverted the change to openssl submodule and fixed grammar in the readme.

**Comment by nicolasnoble** [comment]
It's weird it still says that it can't be merged automatically ? I guess you haven't merged from master yet ?

**Pull request merged**